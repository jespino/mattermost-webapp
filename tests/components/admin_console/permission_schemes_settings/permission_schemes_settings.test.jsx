// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import PermissionSchemesSettings from 'components/admin_console/permission_schemes_settings/permission_schemes_settings.jsx';

describe('components/admin_console/permission_schemes_settings/permission_schemes_settings', () => {
    const defaultProps = {
        schemes: [
            {id: 'id-1', name: 'Test 1', description: 'Test description 1'},
            {id: 'id-2', name: 'Test 2', description: 'Test description 2'},
            {id: 'id-3', name: 'Test 3', description: 'Test description 3'},
        ],
        actions: {
            loadSchemes: jest.fn(() => Promise.resolve([])),
            loadSchemeTeams: jest.fn(),
        },
    };
    test('should match snapshot loading', () => {
        const wrapper = shallow(
            <PermissionSchemesSettings {...defaultProps}/>
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot without schemes', () => {
        const wrapper = shallow(
            <PermissionSchemesSettings
                {...defaultProps}
                schemes={[]}
            />
        );
        wrapper.setState({loading: false});
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot with schemes', () => {
        const wrapper = shallow(
            <PermissionSchemesSettings {...defaultProps}/>
        );
        wrapper.setState({loading: false});
        expect(wrapper).toMatchSnapshot();
    });
});
