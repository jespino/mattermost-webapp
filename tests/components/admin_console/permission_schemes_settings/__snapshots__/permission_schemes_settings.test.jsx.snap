// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/admin_console/permission_schemes_settings/permission_schemes_settings should match snapshot on render 1`] = `
<div
  className="wrapper--fixed"
>
  <h3
    className="admin-console-header"
  >
    <FormattedMessage
      defaultMessage="Permission Schemes"
      id="admin.permissions.permissionSchemes"
      values={Object {}}
    />
  </h3>
  <div
    className="banner info"
  >
    <div
      className="banner__content"
    >
      <span>
        <FormattedMessage
          defaultMessage="Permission Schemes set the default permissions for Team Admins, Channel Admins and everyone else. Learn more about permission schemes in our documentation."
          id="admin.permissions.introBanner"
          values={Object {}}
        />
      </span>
    </div>
  </div>
  <div
    className="permissions-block"
  >
    <div
      className="header"
    >
      <div>
        <h3>
          <FormattedMessage
            defaultMessage="System Scheme"
            id="admin.permissions.systemSchemeBannerTitle"
            values={Object {}}
          />
        </h3>
        <span>
          <FormattedMessage
            defaultMessage="Set the default permissions inherited by all teams unless a Team Override Scheme is applied."
            id="admin.permissions.systemSchemeBannerText"
            values={Object {}}
          />
        </span>
      </div>
      <div
        className="button"
      >
        <Link
          className="btn btn-primary"
          replace={false}
          to="/admin_console/permissions/system-scheme"
        >
          <FormattedMessage
            defaultMessage="Edit Scheme"
            id="admin.permissions.systemSchemeBannerButton"
            values={Object {}}
          />
        </Link>
      </div>
    </div>
  </div>
</div>
`;
